### PR TITLE
Update authorization and authentication flow details + change component name to "security"

### PR DESCRIPTION
**Pull request by tomekpapiernik**
**Description**

Changes proposed in this pull request:

- Authorization and authentication flow details updated to reflect the current state as outlined in [this](https://github.com/kyma-project/community/blob/master/sig-and-wg/sig-core/proposals/authorization-proxy.md) proposal.
- Component name changed to "Security"
- All files related to the documentation image generation updated to use the new component name. 

**Related issue(s)**
#887

### CODE REVIEW

File: docs/manifest.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
       id: tracing
     - displayName: API Gateway
       id: api-gateway
-    - displayName: Authorization and Authentication
-      id: authorization-and-authentication
+    - displayName: Security
+      id: security
```
Comment (kazydek) [inline docs/manifest.yaml]: L.G. suggested to change the order of the components and move the Security topic at the top of the components' list - to important to be at the bottom.

File: docs/security/docs/003-architecture-auth.md [hunk 2]
```diff
@@ -4,11 +4,12 @@ title: Architecture
...
 2. Dex lists all defined Identity Provider connectors to the user. The user selects the Identity Provider to authenticate with. After successful authentication, the browser is redirected back to the OIDC provider which issues a JWT token to the user. After obtaining the token, the browser is redirected back to the Console UI. The Console UI stores the token in the Session Storage and uses it for all subsequent requests.
-3. The Console UI requests for a list of cluster resources in Environments from the API Server. The API Server is not accessible directly. The request is routed through the API Server Proxy - a simple Nginx reverse proxy exposed through an Istio Ingress.
-4. The request arrives at the Kubernetes API Server. The Kubernetes API Server validates the JWT token it received and directs the request accordingly if the validation is successful.
+3. The Authorization Proxy validates the JWT token passed in the `Authorization Bearer` request header. It extracts the user and groups details, the requested resource path, and the request method from the token. The Proxy uses this data to build an attributes record, which it sends to the Kubernetes Authorization API.
```
Comment (kazydek) [inline docs/security/docs/003-architecture-auth.md]: `**Authorization Bearer** request header`

**Comment by tomekpapiernik** [comment]
Not able to merge due to CI failing over and over.

**Comment by tomekpapiernik** [comment]
**Blocked** - the CI keeps failing.

**Pull request merged**